test/unit/org/apache/cassandra/index/sai/cql/DropIndexWhileQueryingTest.java

```diff
@@ -62,13 +98,18 @@ public void testFallbackToAnotherIndex() throws Throwable
         String indexName2 = createIndex("CREATE CUSTOM INDEX ON %s(z) USING 'StorageAttachedIndex'");
         waitForTableIndexesQueryable();
 
+        // TODO this isn't the right place to drop the iterator when we have the histograms. getQueryView
+        // might be better but it might get called at the wrong time and might not lead to the right
+        // coverage for the fallback logic this test is meant to cover.
```

Comment:
@pkolaczk - can you take a look here and see what might make sense independent of SAI version? The original issue was the different time where we would initialize the search iterators.

Reply:
@pkolaczk - how should we update this test to make sure it covers the expected cases for each version?

Reply:
Test improvement merged. LGTM